File: LeetCode/Easy/0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        prev2 = 0
        prev1 = 0

        for i in range(2, len(cost) + 1):
            prev2, prev1 = prev1, min(prev1 + cost[i - 1], prev2 + cost[i - 2])

        return prev1

File: LeetCode/Easy/0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.cs
public class Solution {
    public int MinCostClimbingStairs(int[] cost) {
        var prev2 = 0;
        var prev1 = 0;

        for (var i = 2; i < cost.Length + 1; i++)
        {
            (prev2, prev1) = (prev1, Math.Min(prev1 + cost[i - 1], prev2 + cost[i - 2]));
        }

        return prev1;
    }
}